src/account/memory.js

```diff
@@ -49,16 +50,16 @@ export default AccountBase.compose({
       return
     }
 
-    if (!keypair || typeof keypair !== 'object') throw new Error('KeyPair must be an object')
-    if (!keypair.secretKey || !keypair.publicKey) throw new Error('KeyPair must must have "secretKey", "publicKey" properties')
-    if (typeof keypair.publicKey !== 'string' || keypair.publicKey.indexOf('ak_') === -1) throw new Error('Public Key must be a base58c string with "ak_" prefix')
+    if (!keypair || typeof keypair !== 'object') throw new InvalidKeypairError('KeyPair must be an object')
+    if (!keypair.secretKey || !keypair.publicKey) throw new InvalidKeypairError('KeyPair must must have "secretKey", "publicKey" properties')
+    if (typeof keypair.publicKey !== 'string' || keypair.publicKey.indexOf('ak_') === -1) throw new InvalidKeypairError('Public Key must be a base58c string with "ak_" prefix')
```

Comment:
Later I would propose to accept only private key here, then we don't need most of this validation, the interface would be simpler

Reply:
and generate a public key from the private key?

Reply:
yep, anyway, currently we are doing it for validation